provider/resources.go

```diff
@@ -369,6 +387,23 @@ func preConfigureCallbackWithLogger(ctx context.Context, host *provider.HostClie
 		return fmt.Errorf(
 			"failed to load application credentials.\nTo use your default gcloud credentials, run:\n\t`gcloud auth application-default login`\nSee https://www.pulumi.com/registry/packages/gcp/installation-configuration/ for details.\nExpanded error message: %w", err)
 	}
+
+	if config.Region != "" && config.Project != "" {
+		regionList, err := getRegionsList(config.Project)
+		if err != nil {
+			return fmt.Errorf("failed to get regions list: %w", err)
+		}
+		for _, region := range regionList {
+			if region == config.Region {
+				return nil
+			}
+		}
+		return fmt.Errorf(
+			"region %q is not available for project %q. Available regions: %q",
+			config.Region, config.Project, regionList,
+		)
```

Comment:
We can do better then `%q` for a common user facing error. 
```suggestion
		return fmt.Errorf(
			"region %q is not available for project %q. Available regions: %s",
			config.Region, config.Project, strings.Join(regionList, ", "),
		)
```